src/checkout-buttons/strategies/braintree/braintree-paypal-button-strategy.ts

```diff
@@ -60,32 +62,38 @@ export default class BraintreePaypalButtonStrategy implements CheckoutButtonStra
     }
 
     renderButtons(paypalCheckoutInstance: PaypalClientInstance) {
-        const allowedSources = [];
-        const disallowedSources = [];
         const { paypalOptions, paymentMethod, container } = this._renderButtonsData as RenderButtonsData;
         const { paypal } = this._window;
 
         if (paypal) {
-            if (paypalOptions.allowCredit) {
-                allowedSources.push(paypal.FUNDING.CREDIT);
-            } else {
-                disallowedSources.push(paypal.FUNDING.CREDIT);
+            const FUNDING_SOURCES = [];
+            for (const fundingKey in paypal.FUNDING) {
+                if (fundingKey !== 'CARD') {
+                    if ( fundingKey === 'CREDIT' && !paypalOptions.allowCredit) {
```

Comment:
Could you rewrite this to avoid nested ifs please?

Reply:
fixed